fractal_server/app/routes/admin/v2.py

```diff
@@ -307,3 +314,126 @@ async def flag_task_v1_as_v2_compatible(
+
+    task_v2_minimal: TaskV2Minimal
+
+    workflow_id: Optional[int]
+    workflow_name: Optional[str]
+    project_id: Optional[int]
+    project_name: Optional[str]
+
+    project_users: Optional[list[ProjectUser]]
+
+
+@router_admin_v2.get("/task/", response_model=list[TaskContext])
+async def query_tasks(
+    id: Optional[int] = None,
+    source: Optional[str] = None,
+    version: Optional[str] = None,
+    name: Optional[str] = None,
+    owner: Optional[str] = None,
+    only_no_owner: bool = False,
+    max_number_of_results: int = 25,
+    user: User = Depends(current_active_superuser),
+    db: AsyncSession = Depends(get_async_db),
+) -> list[TaskContext]:
+
+    if (owner is not None) and (only_no_owner is True):
+        raise HTTPException(
+            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            detail="Cannot query Tasks by owner if `only_no_owner` is True",
+        )
+
+    stm = select(TaskV2)
+
+    if id is not None:
+        stm = stm.where(TaskV2.id == id)
+    if source is not None:
+        stm = stm.where(TaskV2.source == source)
+    if version is not None:
+        stm = stm.where(TaskV2.version == version)
+    if name is not None:
+        stm = stm.where(TaskV2.name.icontains(name))
+
+    if owner is not None:
+        stm = stm.where(TaskV2.owner == owner)
+    elif only_no_owner is True:
+        stm = stm.where(TaskV2.owner == None)  # noqa E711
+
+    res = await db.execute(stm)
+    task_list = res.scalars().all()
+    if len(task_list) > max_number_of_results:
+        await db.close()
+        raise HTTPException(
+            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            detail=(
+                f"Too many Tasks ({len(task_list)} > {max_number_of_results})."
+                " Please add more query filters."
+            ),
+        )
+
+    contexts = []
+
+    for task in task_list:
+        stm = (
+            select(WorkflowV2)
+            .join(WorkflowTaskV2)
+            .where(WorkflowTaskV2.workflow_id == WorkflowV2.id)
+            .where(WorkflowTaskV2.task_id == task.id)
+        )
+        res = await db.execute(stm)
+        wf_list = res.scalars().all()
+
+        if wf_list:
+            for workflow in wf_list:
+                contexts.append(
+                    TaskContext(
+                        task_v2_minimal=TaskV2Minimal(**task.model_dump()),
+                        workflow_id=workflow.id,
+                        workflow_name=workflow.name,
+                        project_id=workflow.project.id,
+                        project_name=workflow.project.name,
+                        project_users=[
+                            ProjectUser(id=user.id, email=user.email)
+                            for user in workflow.project.user_list
+                        ],
+                    )
+                )
+        else:
+            contexts.append(
+                TaskContext(task_v2_minimal=TaskV2Minimal(**task.model_dump()))
```

Comment:
Same as above:

There is no real need of constructing Pydantic models here, since we are validating the same response body twice (here, and when the endpoint ends). I think we can stick with `dict`, within the endpoint, unless we expect some validation to be relevant for some specific reason..